### PR TITLE
Optionally disable logging in the data sampler to support predict_step

### PR DESCRIPTION
**Pull request by jstjohn**
Pytorch lightning's `predict_step` does not support logging on the module. This PR adds an option to the data sampler to disable logging which allows the predict step to work.

### CODE REVIEW

**Comment by ko3n1g** [comment]
stopping this for now until our CI is out of maintenance

File: nemo/lightning/pytorch/plugins/data_sampler.py [hunk 1]
```diff
@@ -94,28 +96,33 @@ def on_megatron_step_end(self, trainer: pl.Trainer, pl_module: pl.LightningModul
...
-        consumed_samples = self.compute_consumed_samples(trainer.global_step + 1 - self.init_global_step)
-        pl_module.log(
-            'consumed_samples',
-            consumed_samples,
-            prog_bar=True,
-            rank_zero_only=True,
-            batch_size=1,
-        )
+        # TODO: Add consumed samples
+        consumed_samples = self.compute_consumed_samples(trainer.global_step + 1 - self.init_consumed_samples)
```
Comment (jstjohn) [inline nemo/lightning/pytorch/plugins/data_sampler.py:100]: I'm going to remove this diff

**Comment by ashors1** [comment]
Looks good, thanks! Can we clean up some of the old comments in the test script?

**Review by ashors1 — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**